guides/04_integrating-other-frameworks/Gradio-and_Comet.md

```diff
@@ -0,0 +1,273 @@
+
+```shell
+export COMET_API_KEY="<Your API Key>"
+export COMET_WORKSPACE="<Your Workspace Name>"
+export COMET_PROJECT_NAME="<Your Project Name>"
+```
+
+or set them in a `.comet.config` file in your working directory. You file should be formatted in the following way.
+
+```shell
+[comet]
+api_key=<Your API Key>
+workspace=<Your Workspace Name>
+project_name=<Your Project Name>
+```
+
+If you are using the provided Colab Notebooks to run these examples, please run the cell with the following snippet before starting the Gradio UI. Running this cell allows you to interactively add your API key to the notebook.
+
+```python
+import comet_ml
+comet_ml.init()
+```
+
+## 1. Logging Gradio UI's to your Comet Experiments
+
+[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/comet-ml/comet-examples/blob/master/integrations/model-evaluation/gradio/notebooks/Gradio_and_Comet.ipynb)
+
+In this example, we will go over how to log your Gradio Applications to Comet and interact with them using the Gradio Custom Panel.
+
+Let's start by building a simple Image Classification example using `resnet18`.
+
+```python
+import comet_ml
+
+import requests
+import torch
+from PIL import Image
+from torchvision import transforms
+
+torch.hub.download_url_to_file("https://github.com/pytorch/hub/raw/master/images/dog.jpg", "dog.jpg")
+
+if torch.cuda.is_available():
+    device = "cuda"
+else:
+    device = "cpu"
+
+model = torch.hub.load("pytorch/vision:v0.6.0", "resnet18", pretrained=True).eval()
+model = model.to(device)
+
+# Download human-readable labels for ImageNet.
+response = requests.get("https://git.io/JJkYN")
+labels = response.text.split("\n")
+
+
+def predict(inp):
+    inp = Image.fromarray(inp.astype("uint8"), "RGB")
+    inp = transforms.ToTensor()(inp).unsqueeze(0)
+    with torch.no_grad():
+        prediction = torch.nn.functional.softmax(model(inp.to(device))[0], dim=0)
+    return {labels[i]: float(prediction[i]) for i in range(1000)}
+
+
+inputs = gr.Image()
+outputs = gr.Label(num_top_classes=3)
+
+io = gr.Interface(
+    fn=predict, inputs=inputs, outputs=outputs, examples=["dog.jpg"]
+)
+io.launch(inline=False, share=True)
+
+experiment = comet_ml.Experiment()
+experiment.add_tag("image-classifier")
+
+io.integrate(comet_ml=experiment)
+```
+
+The last line in this snippet will log the URL of the Gradio Appication to your Comet Experiment. You can find the URL in the Text Tab of your Experiment.
+
+https://user-images.githubusercontent.com/7529846/214328034-09369d4d-8b94-4c4a-aa3c-25e3ed8394c4.mp4
+
+Add the Gradio Panel to your Experiment to interact with your application.
+
+https://user-images.githubusercontent.com/7529846/214328194-95987f83-c180-4929-9bed-c8a0d3563ed7.mp4
+
+
+## 2. Embedding Gradio Applications directly into your Comet Projects
+
+<iframe src="https://youtu.be/KZnpH7msPq0?t=7" frameborder="0"
```

Comment:
I don't think this embed link is working:

![image](https://user-images.githubusercontent.com/41651716/215873535-accdb993-3e5a-4460-bb60-26edb18e61e7.png)


I think we should use the one created by youtube from the video itself. See here:

![image](https://user-images.githubusercontent.com/41651716/215873475-5591713f-5caa-4415-8196-52322db6b69e.png)